python/samples/concepts/auto_function_calling/function_calling_with_required_type.py

```diff
@@ -0,0 +1,185 @@
+# The value may be configured to be more than one depending upon your scenario.
+execution_settings = OpenAIChatPromptExecutionSettings(
+    service_id=service_id,
+    max_tokens=2000,
+    temperature=0.7,
+    top_p=0.8,
+    function_choice_behavior=FunctionChoiceBehavior.Required(filters={"included_plugins": ["math", "time"]}),
+)
+
+history = ChatHistory()
+
+history.add_system_message(system_message)
+history.add_user_message("Hi there, who are you?")
+history.add_assistant_message("I am Mosscap, a chat bot. I'm trying to figure out what people need.")
+
+arguments = KernelArguments(settings=execution_settings)
+
+
+def print_tool_calls(message: ChatMessageContent) -> None:
+    # A helper method to pretty print the tool calls from the message.
+    # This is only triggered if auto invoke tool calls is disabled.
+    items = message.items
+    formatted_tool_calls = []
+    for i, item in enumerate(items, start=1):
+        if isinstance(item, FunctionCallContent):
+            tool_call_id = item.id
+            function_name = item.name
+            function_arguments = item.arguments
+            formatted_str = (
+                f"tool_call {i} id: {tool_call_id}\n"
+                f"tool_call {i} function name: {function_name}\n"
+                f"tool_call {i} arguments: {function_arguments}"
+            )
+            formatted_tool_calls.append(formatted_str)
+    print("Tool calls:\n" + "\n\n".join(formatted_tool_calls))
+
+
+async def handle_streaming(
+    kernel: Kernel,
+    chat_function: "KernelFunction",
+    arguments: KernelArguments,
+) -> None:
+    response = kernel.invoke_stream(
+        chat_function,
+        return_function_results=False,
+        arguments=arguments,
+    )
+
+    print("Mosscap:> ", end="")
+    streamed_chunks: list[StreamingChatMessageContent] = []
+    async for message in response:
+        if not execution_settings.function_choice_behavior.auto_invoke_kernel_functions and isinstance(
+            message[0], StreamingChatMessageContent
+        ):
+            streamed_chunks.append(message[0])
+        else:
+            print(str(message[0]), end="")
+
+    if streamed_chunks:
+        streaming_chat_message = reduce(lambda first, second: first + second, streamed_chunks)
+        print("Auto tool calls is disabled, printing returned tool calls...")
+        print_tool_calls(streaming_chat_message)
+
+    print("\n")
+
+
+async def chat() -> bool:
+    try:
+        user_input = input("User:> ")
+    except KeyboardInterrupt:
+        print("\n\nExiting chat...")
+        return False
+    except EOFError:
+        print("\n\nExiting chat...")
+        return False
+
+    if user_input == "exit":
+        print("\n\nExiting chat...")
+        return False
+    arguments["user_input"] = user_input
+    arguments["chat_history"] = history
+
+    stream = False
+    if stream:
+        await handle_streaming(kernel, chat_function, arguments=arguments)
```

Comment:
This concept example, when configured with `Required` tool choice and streaming fails to produce a result. Currently investigating.